R/utilities.R

```diff
@@ -53,14 +53,25 @@ has_class <- function(x, cls) {
 #' ```
 #'
 #' @param vars an `ivars` object as created by [set_vars()]
+#' @param is_multi_groups logical. If `TRUE`, the `vars$group` variable is added in the model formula.
+#' It should be `TRUE` if `vars$group` has at least 2 levels, `FALSE` otherwise.
 #' @return
 #' A formula
-as_simple_formula <- function(vars) {
-    variables <- c(
-        vars$group,
-        vars$visit,
-        vars$covariates
+as_simple_formula <- function(vars, is_multi_groups) {
+
+    variables <- ife(
+        is_multi_groups,
+        c(
+            vars$group,
+            vars$visit,
+            vars$covariates
+        ),
+        c(
+            vars$visit,
+            vars$covariates
+        )
```

Comment:
I'm wondering if there is a cleaner way to do this.  Like for example if its a single group could we not just set `vars$group` to NULL ?  This would save having to pass this `is_multi_groups` option all over the place :(

Reply:
True. In my opinion it does not change that much. The current solution I would say is conceptually cleaner since the part related to the fixed effect formula is all in the same function instead of adapting the formula outside it. However yes, it needs this additional `is_multi_groups` argument. Let me know what you prefer, I am inclined to say that it does not matter, but I am happy to change if you think it is better.